Jenkinsfile

```diff
@@ -77,16 +77,18 @@ def cleanup_master() {
 }
 
 def archive_logs() {
-	sh """
-	cd "\$(echo ${env.WORKSPACE} | cut -f 1 -d '@')"
-	LOGS_NAME=\$(echo "logs-$JOB_NAME-$NODE_NAME-$BUILD_ID" | sed 's/\\///g')
-	mv logs "\$LOGS_NAME"
+	sh '''
+	cd "$(echo "$WORKSPACE" | cut -f 1 -d '@')"
+	# need a unique and descriptive name so that the artifacts don't clobber each other
+	# and we can know which log file goes to which node, etc
+	LOGS_NAME=$(echo "logs-$JOB_NAME-$NODE_NAME-$BUILD_ID" | sed 's/\\///g')
+	mv logs "$LOGS_NAME"
 
-	if [ "\$(pwd)" != "${env.WORKSPACE}" ]; then
-		rm -r "${env.WORKSPACE}"/logs* || true
-		cp -r \$LOGS_NAME ${env.WORKSPACE}
+	if [ "$(pwd)" != "$WORKSPACE" ]; then
```

Comment:
What's happening here?

Reply:
run_action has similar logic sort of.
sometimes, you could end up in an "...@2" workspace, but all of the lisk stuff is in the base workspace name - archiveArtifacts reads from the "real" workspace though. So this makes sure the logs are where archiveArtifacts can read them